### PR TITLE
Add clearer docs around coroutine support

### PR DESCRIPTION
**Pull request by scprek**
I had a hard time figuring out how to propagate otel MDC data into coroutines so adding more documentation here.

Also, I'm not sure what opentelemetry-extension-kotlin-1.0 in the instrumentation directory is or differs?

### CODE REVIEW

**Comment by scprek** [comment]
sorry about the spotless, I don't know what's going on with this project on my machine. I don't have any gradle issues in other projects.

File: instrumentation/kotlinx-coroutines/README.md
Comment (mateuszrzeszutek) [inline instrumentation/kotlinx-coroutines/README.md:1]: I think you should add this doc to the Kotlin extension in the `opentelemetry-java` repo instead. This project simply uses that extension to add coroutine support for the javaagent users; the manual usage should be documented where the code lives.

Comment (scprek) [reply instrumentation/kotlinx-coroutines/README.md:1]: I had the same thought, will do

Comment (mateuszrzeszutek) [reply instrumentation/kotlinx-coroutines/README.md:1]: Thanks!

**Comment by breedx-splk** [comment]
@scprek Any updates on this? Still going to add the doc to https://github.com/open-telemetry/opentelemetry-java/tree/main/extensions/kotlin and remove it from this PR? Thanks!

**Comment by scprek** [comment]
Just busy with other stuff but yeah can close this and will move at some point

**Comment by breedx-splk** [comment]
Thanks for the quick reply. No pressure, just circling back. I'll leave it open so that we don't lose sight of it. Thanks!

File: instrumentation/kotlinx-coroutines/README.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
 # Kotlin Coroutines
 
+[opentelemetry-java](https://github.com/open-telemetry/opentelemetry-java) has various plugins mentioned in their [README](https://github.com/open-telemetry/opentelemetry-java/tree/main#api-extensions)
+
+To enable coroutine support, you need to add the `io.opentelemetry:opentelemetry-extension-kotlin` dependency.
+
+Then you can do:
+
+```kotlin
+launch(Context.current().asContextElement()) {
```
Comment (CraftyFella) [inline instrumentation/kotlinx-coroutines/README.md:10]: This is gold and saved me many hours..

File: instrumentation/kotlinx-coroutines/README.md
Comment (trask) [reply instrumentation/kotlinx-coroutines/README.md:1]: @scprek just checking in on this PR, thx!

**Comment by scprek** [comment]
Moved to https://github.com/open-telemetry/opentelemetry-java/pull/5799 if you'd like to see it

**Pull request closed without merging**